jenkins/bootstrap.py

```diff
@@ -735,11 +737,16 @@ def job_args(args):
     return [os.path.expandvars(a) for a in args]
 
 
-def job_script(job, use_json):
+def job_script(job, use_json, jobs_dir):
     """Return path to script for job."""
+    jobs_dir_resolver = lambda f: test_infra('jobs/%s' % f)
+
+    if jobs_dir:
+        jobs_dir_resolver = lambda f: os.path.join(jobs_dir, f)
+
     if not use_json:
```

Comment:
Our `config.yaml` is [here](https://github.com/istio/test-infra/blob/master/prow/config.yaml). It runs an image with an entrypoint [here](https://github.com/istio/test-infra/blob/master/docker/prowbazel/entrypoint). Notice the image does run bootstrap, just without some of the k8s env stuff and with a new flag specifying a directory to find the job in (the effect of the `--no-magic-env` and `--jobs-dir` flags -- the original names for the flags introduced in this PR).

So the job `.sh` files themselves live in the respective repositories. A simple example in the istio/test-infra repo is the presubmit job file [here](https://github.com/istio/test-infra/blob/master/prow/test-infra-presubmit.sh). The job files always live in the `prow` directory of the target repo (name `prow` is convention). So for jobs against the istio/pilot repo, they live [here](https://github.com/istio/pilot/tree/master/prow)

a few moving parts here so follow up if i left something unclear.